### PR TITLE
Adds common string parser class with parse helper methods

### PR DESCRIPTION
**Pull request by wieslawsoltes**
#### Reference Issue

Split from #786

#### What does this implement/fix? Explain your changes.

Adds common string parser class with parse helper methods.

### CODE REVIEW

**Comment by wieslawsoltes** [comment]
@mrbean-bremen Can we do quick review of this PR, this is necessary to split the rest of #786 PR

**Review by mrbean-bremen — APPROVED** [review]

**Comment by H1Gdev** [comment]
@wieslawsoltes 
@mrbean-bremen 
What are rules for using these helper methods ?

File: Source/Helpers/StringParser.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace Svg.Helpers
+{
+    internal static class StringParser
+    {
+        private static readonly CultureInfo Format = CultureInfo.InvariantCulture;
+
+        public static float ToFloat(ref ReadOnlySpan<char> value)
+        {
+#if NETSTANDARD2_1 || NETCORE || NETCOREAPP2_1 || NETCOREAPP3_1 || NET5_0
```
Comment (H1Gdev) [inline Source/Helpers/StringParser.cs:12]: Considering the future, I think that the following will improve maintainability.
```cs
#if Net4
    return float.Parse(value.ToString(), NumberStyles.Float, Format);
#else
    return float.Parse(value, NumberStyles.Float, Format);
#endif
```

**Comment by H1Gdev** [comment]
@wieslawsoltes 
What do we need to be aware of in terms of performance ? (Method ? Arguments ?)

**Comment by wieslawsoltes** [comment]
> @wieslawsoltes
> @mrbean-bremen
> What are rules for using these helper methods ?

See the usage examples https://github.com/svg-net/SVG/pull/786/files

They follow original code usage.

**Comment by wieslawsoltes** [comment]
> @wieslawsoltes
> What do we need to be aware of in terms of performance ? (Method ? Arguments ?)

Do not allocate new strings e.g." "string".Trim() allocates, use "string".AsSpan() etc. see the #786 PR

File: Source/Helpers/StringParser.cs (hunk 1 above)
Comment (wieslawsoltes) [reply Source/Helpers/StringParser.cs:12]: NETSTANDARD2_0 does not support that API

Comment (H1Gdev) [reply Source/Helpers/StringParser.cs:12]: @wieslawsoltes 
Add NETSTANDARD2_0, too.
The newly added API will not be removed in the future.
However, as it is, it has to be added to `#if` every time the version of Framework goes up.

Comment (wieslawsoltes) [reply Source/Helpers/StringParser.cs:12]: It requires netstandard2.1

Comment (H1Gdev) [reply Source/Helpers/StringParser.cs:12]:
```cs
#if Net4 || NETSTANDARD2_0
    return float.Parse(value.ToString(), NumberStyles.Float, Format);
#else
    return float.Parse(value, NumberStyles.Float, Format);
#endif
```

Comment (wieslawsoltes) [reply Source/Helpers/StringParser.cs:12]: Now yes.

File: Source/Helpers/StringSplitEnumerator.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ public bool MoveNext()
                     return true;
                 }
 
-                var slice = span.Slice(0, index).Trim();
+                var slice = span.Slice(0, index);
                 _str = span.Slice(index + 1);
                 if (slice.Length == 0)
```
Comment (H1Gdev) [inline Source/Helpers/StringSplitEnumerator.cs:40]: https://github.com/svg-net/SVG/blob/5dc4460eaf9819909b8a3beb993f3ba4eb7dbf6e/Source/Helpers/StringSplitEnumerator.cs#L38-L44

The following code is more efficient.
```cs
    _str = span.Slice(index + 1);
    if (index == 0)
        continue;
    Current = new StringPart(span.Slice(0, index));
```

**Pull request merged**